### PR TITLE
PEAR-1838 - Annotations Browser main page

### PR DESCRIPTION
**Pull request by amlehman**
## Description
\+ small refactor to make facet code more generic/reusable

## Checklist

- [x] Added proper unit tests
- [x] Left proper TODO messages for any remaining tasks
- [x] Scanned for web accessibility with **aXe**, and mitigated or documented
      flagged issues

## Screenshots/Screen Recordings (if Appropriate)
Screenshot 2024-03-15 at 11 06 22 AM

### CODE REVIEW

**Comment by wteouchicago** [comment]
Looking good.

Issues:
1. "Category" is missing as a table column. It should be left of Classification, displayed by default, and sortable
2. Any chance you can make the "Total of X Annotations" above the table similar  to what we have in the Repository in its style and position, e.g. against the search field? I think @cemile-ctds intends to make all the "Total of X things" text that style eventually (Cemile, please confirm).
3.  Not really an issue, but my eyes/brain are feeling some uneasiness looking at the last card on the left, but I'm not sure why. _Maybe_ it's the amount of white space below the content of the card? Maybe @cemile-ctds knows what I mean.

Screenshot 2024-03-14 at 10 20 31 PM

**Comment by amlehman** [comment]
Hi, @wteouchicago. I addressed 1 & 2. I'll wait to see if Cemile has any feedback on 3.

**Comment by cemile-ctds** [comment]
2. Thank you. This looks good.
3. The spacing appears to be good at the moment, but I noticed that the border color of the last facet card is lighter than the others.
Screenshot 2024-03-15 at 12 36 26

-------
Screenshot 2024-03-15 at 12 35 55

4. When the user is in "Browse Annotations," please ensure that the button is in an active state, similar to the "Manage Sets" button.
5. Please add a heading to the page, like "Manage Sets." I'll leave the copy to @wteouchicago.

**Comment by wteouchicago** [comment]
the heading can be "Browse Annotations" for now (I may change to "Annotations" or "Annotations Browser" later).

**Comment by cemile-ctds** [comment]
This looks good on my side :)

**Comment by wteouchicago** [comment]
looks good to me too. 👍  even issue 3 looks less unsettling now that the border color has changed.

File: packages/portal-proto/src/features/annotations/registerApp.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  App: AnnotationsBrowser,
+  id: id,
+  name: "Annotations Browser",
+  version: "v1.0.0",
+  requiredEntityTypes: [],
+  store: AppStore,
+  context: AppContext,
+});
+
+export const AnnotationBrowserAppId: string = id;
```
Comment (wteouchicago) [inline packages/portal-proto/src/features/annotations/registerApp.ts:15]: do we need to make any changes here now that it's been un-apped? @amlehman 
I saw some message about:

> portal-proto: Module not found: Can't resolve './AnnotationsBrowser' in '/Users/wendyteo/code/gdc-frontend-framework/packages/portal-proto/src/features/apps'

Comment (amlehman) [reply packages/portal-proto/src/features/annotations/registerApp.ts:15]: Where are you seeing that message @wteouchicago ? I'm not seeing it in the console or the build.

Comment (wteouchicago) [reply packages/portal-proto/src/features/annotations/registerApp.ts:15]: @amlehman 😵 it showed up when I ran `npm run dev` but it's gone now after I switched branches and back. false alarm?

**Review by craigrbarnes — APPROVED** [review]
Implementation as anapp is fine. 
The refactor of the "project" query special handling is a nice improvement.

**Pull request merged**